### PR TITLE
Partly move setup.py config to pyproject.toml

### PR DESCRIPTION
**Pull request by ggravlingen**
#### Reason for change
This is an attempt of nudging [this issue](https://github.com/Arelle/Arelle/issues/285) a bit forward.

#### Description of change
This PR:
- Cleans up whitespace
- Moves configuration to `pyproject.toml` from `setup.py`
- Adds a version file. If you want to continue using dynamic versions from Github tag, that looks quite possible using the `dynamic` metadata: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata.

#### Steps to Test
I installed the package locally using `python3 -m pip install -e .`

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by sagesmith-wf — APPROVED** [review]

**Comment by ggravlingen** [comment]
I wasn't 100% sure but I think these can also be removed now?
https://github.com/Arelle/Arelle/blob/75182094e068ac685545bec538e73754c4f3a5f6/setup.py#L29
https://github.com/Arelle/Arelle/blob/75182094e068ac685545bec538e73754c4f3a5f6/setup.py#L32

File: pyproject.toml [hunk 1]
```diff
@@ -2,6 +2,69 @@
 requires = ["setuptools~=62.3", "wheel~=0.37.1"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "arelle-release"
```
Comment (ggravlingen) [inline pyproject.toml:6]: Looks like version is required here:
```
DEFINITION:
    {
        "required": [
            "version"
        ]
    }

```

Comment (ggravlingen) [reply pyproject.toml:6]: Maybe we can consume `__version__` in `Version.py` by using the dynamic method?

**Review by sagesmith-wf — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (austinmatherne-wk) [reply pyproject.toml:6]: Damn, okay. `__version__` is the best option for now, go with that.

**Review by sagesmith-wf — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]
Verified python package along with confirming cx freeze builds are not impacted.

**Pull request merged**